Classes/Util.php

```diff
@@ -274,22 +274,23 @@ public static function getConfigurationFromPageId(
 
         if ($initializeTsfe) {
             self::initializeTsfe($pageId, $language);
-            if (!empty($configurationToUse)) {
-                return self::buildTypoScriptConfigurationFromArray($configurationToUse, $pageId, $language, $path);
+            if (empty($configurationToUse)) {
+                $configurationToUse = self::getConfigurationFromInitializedTSFE($path);
             }
-            $configurationToUse = self::getConfigurationFromInitializedTSFE($path);
         } else {
-            if (!empty($configurationToUse)) {
-                return self::buildTypoScriptConfigurationFromArray($configurationToUse, $pageId, $language, $path);
+            if (empty($configurationToUse)) {
+                $configurationToUse = self::getConfigurationFromExistingTSFE($pageId, $path, $language);
```

Comment:
Hrmm - might be the case - but most calls of the method `getConfigurationFromPageId ` are called with `initializeTsfe = false` and thus it should have been a "problem" before also. I'll leave it up to @timohund to decide what should be done in this case